src/targets/gpu/kernels/include/migraphx/kernels/bit_cast.hpp

```diff
@@ -23,15 +23,31 @@
 #define MIGRAPHX_GUARD_KERNELS_BITCAST_HPP
 
 #include <migraphx/kernels/type_traits.hpp>
+#include <migraphx/kernels/vec.hpp>
 
 namespace migraphx {
+
 template <typename To,
           typename From,
+          MIGRAPHX_REQUIRES(is_any_vec<To>()),
           MIGRAPHX_REQUIRES(is_trivially_copyable<To>{} and is_trivially_copyable<From>{})>
 inline constexpr To bit_cast(From fr) noexcept
+{
```

Comment:
This function should return `auto`. `To` is not valid return with vector types. If you do `bit_cast<int8_t>(vec<uint8_t, 2>{})` it should return `vec<int8_t, 2>{}` not `int8_t`. The `vec_transform` functor already takes care of figuring out the return type for you, so you can just return `auto` instead.

Reply:
Same issue. When compiling float8 for GPU this version is used.

Reply:
Same issue? The return type is wrong for this. If you bit_cast a vector type you should get a vector type, not the scalar type.

Reply:
The issue comes from the usage here: https://github.com/ROCm/AMDMIGraphX/blob/cd37826b9f2137cc7fa5f57a4e37437d09bb1daf/src/targets/gpu/kernels/include/migraphx/kernels/float8_impl.hpp#L268

Reply:
That takes a scalar input not a vector input.

Reply:
You were right, changing the return type to `auto` resolves the issue. Thanks, made the fix.